libraries/Servo/Servo.cpp

```diff
@@ -0,0 +1,305 @@
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#include <Arduino.h>
+#include <Servo.h>
+
+// Converts microseconds to timer tick and vice versa
+#define usToTicks(_us)     ((TMR_FREQ / 1000000L) * (_us))
+#define ticksToUs(_ticks)  ((unsigned)_ticks / (TMR_FREQ / 1000000L))
+
+#define TRIM_DURATION  5              // compensation ticks to trim adjust for digitalWrite delays
+
+static servo_t servos[MAX_SERVOS];    // static array of servo structures
+
+static uint8_t ServoCount = 0;        // the total number of attached servos
+
+// Index for the servo being pulsed for each timer (or -1 if refresh interval)
+static volatile int8_t currentServoIndex[_Nbr_16timers];
+#define REFRESH_INTERVAL_PERIOD            (-1)
+
+// Timer controlling this servo
+#define SERVO_INDEX_TO_TIMER(_servo_nbr)   ((timer16_Sequence_t)(_servo_nbr / SERVOS_PER_TIMER))
+// Index of the servo on this timer
+#define SERVO_INDEX_TO_CHANNEL(_servo_nbr) (_servo_nbr % SERVOS_PER_TIMER)
+// Servo index by timer and channel
+#define SERVO_INDEX(_timer, _channel)      ((_timer * SERVOS_PER_TIMER) + _channel)
+// Servo structure by timer and channel
+#define SERVO(_timer, _channel)            (servos[SERVO_INDEX(_timer,_channel)])
+
+/************ static functions common to all instances ***********************/
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+inline void Servo_Handler(timer16_Sequence_t timer, NRF_TIMER_Type *nrfTimer, uint32_t ccReg)
+{
+  // Clear the interrupt event flag
+  nrfTimer->EVENTS_COMPARE[ccReg] = 0;
+
+  if (currentServoIndex[timer] == REFRESH_INTERVAL_PERIOD) {
+    // Refresh interval completed so reset the timer
+    nrfTimer->TASKS_CLEAR = 1;
+    nrfTimer->CC[ccReg] = 0;
+  } else {
+    if (SERVO_INDEX(timer, currentServoIndex[timer]) < ServoCount &&
+        SERVO(timer, currentServoIndex[timer]).Pin.isActive == true) {
+      // It's an active channel so pulse it low
+      NRF_GPIO->OUTCLR = digitalPinToBitMask(SERVO(timer, currentServoIndex[timer]).Pin.nbr);
+    }
+  }
+
+  // Select the next servo controlled by this timer
+  currentServoIndex[timer]++;
+  if (SERVO_INDEX(timer, currentServoIndex[timer]) < ServoCount &&
+      currentServoIndex[timer] < SERVOS_PER_TIMER) {
+    if (SERVO(timer, currentServoIndex[timer]).Pin.isActive == true) {
+      // It's an active channel so pulse it high
+      NRF_GPIO->OUTSET = digitalPinToBitMask(SERVO(timer,currentServoIndex[timer]).Pin.nbr);
+    }
+    // Get the counter value
+    unsigned int ticks = SERVO(timer, currentServoIndex[timer]).ticks;
+    nrfTimer->CC[ccReg] += ticks + (ticks >> 6);
+  } else {
+    // Finished all channels so wait for the refresh period to expire before starting over
+    // Allow a few ticks to ensure the next timer event is not missed
```

Comment:
I am not quite sure exactly why this `+ 4` is done, I've seen it in multiple implementations of the servo library for different platforms, although I assume it's been carried over from the original Arduino implementation.
Is it just to just to give us some head room in case the counter (which keeps running within the interrupt) hits the REFRESH_INTERVAL value? I guess that way we don't miss it and gets trigger very soon after the interrupt?